src/lib/utils/bit_ops.h

```diff
@@ -41,9 +41,7 @@ BOTAN_FORCE_INLINE constexpr T ct_is_zero(T x) {
 */
 template <std::unsigned_integral T>
 BOTAN_FORCE_INLINE constexpr size_t ct_if_is_zero_ret(T x, size_t s) {
-   const T a = ~x & (x - 1);
-   const size_t mask = static_cast<size_t>(0) - static_cast<size_t>(a >> (sizeof(T) * 8 - 1));
-   return mask & s;
+   return ct_is_zero(x) & s;
 }
```

Comment:
Doesn't that cause trouble if `T` is shorter than `size_t`? Then `ct_is_zero()` returns a 1-mask that is shorter than `size_t` which would truncate the MSBs of `s`.

Reply:
#5181